### PR TITLE
feat/Devtooling-459: Added BuildAPIDiagnosticError function calls

### PR DESCRIPTION
**Pull request by BrianMoyles**

### CODE REVIEW

File: genesyscloud/architect_datatable/resource_genesyscloud_architect_datatable.go [hunk 1]
```diff
@@ -49,9 +49,8 @@ func getAllArchitectDatatables(ctx context.Context, clientConfig *platformclient
 
 	archProxy := getArchitectDatatableProxy(clientConfig)
 	tables, resp, err := archProxy.getAllArchitectDatatable(ctx)
-
 	if err != nil {
-		return resources, diag.Errorf("Error encountered while calling getAllArchitectDatattables %s %v.\n", err, resp)
+		return resources, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Error encountered while calling getAllArchitectDatattables"), resp)
```
Comment (charliecon) [inline genesyscloud/architect_datatable/resource_genesyscloud_architect_datatable.go:53]: @carnellj Should we be including the `err` in the summary string?

File: genesyscloud/architect_flow/data_source_genesyscloud_flow.go [hunk 2]
```diff
@@ -22,9 +22,9 @@ func dataSourceFlowRead(ctx context.Context, d *schema.ResourceData, m interface
 	// Query flow by name. Retry in case search has not yet indexed the flow.
 	return util.WithRetries(ctx, 5*time.Second, func() *retry.RetryError {
 		for pageNum := 1; ; pageNum++ {
-			flows, _, getErr := p.GetAllFlows(ctx)
+			flows, resp, getErr := p.GetAllFlows(ctx)
 			if getErr != nil {
-				return retry.NonRetryableError(fmt.Errorf("error requesting flow %s: %s", name, getErr))
+				return retry.NonRetryableError(fmt.Errorf("Error requesting flow %s: %s %s", name, getErr, resp))
```
Comment (charliecon) [inline genesyscloud/architect_flow/data_source_genesyscloud_flow.go:27]: For this line, I see the warning "Error string should not be capitalized or end with punctuation mark" on GoLand. It's fairly non-consequential, but better to keep warnings to a minimum at the same time

File: genesyscloud/architect_schedulegroups/resource_genesyscloud_architect_schedulegroups.go [hunk 3]
```diff
@@ -28,7 +28,7 @@ func getAllAuthArchitectSchedulegroups(ctx context.Context, clientConfig *platfo
 
 	scheduleGroups, proxyResponse, err := proxy.getAllArchitectSchedulegroups(ctx)
 	if err != nil {
-		return nil, diag.Errorf("Failed to get page of schedule groups: %v %v", err, proxyResponse)
+		return nil, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to get page of schedule groups"), proxyResponse)
```
Comment (charliecon) [inline genesyscloud/architect_schedulegroups/resource_genesyscloud_architect_schedulegroups.go:31]: missing the error from this string

File: genesyscloud/architect_schedulegroups/resource_genesyscloud_architect_schedulegroups.go [hunk 4]
```diff
@@ -52,7 +52,7 @@ func createArchitectSchedulegroups(ctx context.Context, d *schema.ResourceData,
 		if strings.Contains(fmt.Sprintf("%v", err), "routing:schedule:add") {
 			msg = "\nYou must have all divisions and future divisions selected in your OAuth client role"
 		}
-		return diag.Errorf("Failed to create schedule group %s | ERROR: %s%s %v", *schedGroup.Name, err, msg, proxyResponse)
+		return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to create schedule group %s %s", d.Id(), msg), proxyResponse)
```
Comment (charliecon) [inline genesyscloud/architect_schedulegroups/resource_genesyscloud_architect_schedulegroups.go:55]: the previous error message included `err` as well as `msg`

File: genesyscloud/architect_schedulegroups/resource_genesyscloud_architect_schedulegroups.go [hunk 5]
```diff
@@ -128,7 +128,7 @@ func updateArchitectSchedulegroups(ctx context.Context, d *schema.ResourceData,
 		if strings.Contains(fmt.Sprintf("%v", err), "routing:schedule:add") {
 			msg = "\nYou must have all divisions and future divisions selected in your OAuth client role"
 		}
-		return diag.Errorf("Failed to update schedule group %s: %s%s %v", d.Id(), err, msg, proxyResponse)
+		return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to update schedule group %s. %s", d.Id(), msg), proxyResponse)
```
Comment (charliecon) [inline genesyscloud/architect_schedulegroups/resource_genesyscloud_architect_schedulegroups.go:131]: Same here. Missing `err`

File: genesyscloud/architect_schedulegroups/resource_genesyscloud_architect_schedulegroups.go [hunk 6]
```diff
@@ -146,7 +146,7 @@ func deleteArchitectSchedulegroups(ctx context.Context, d *schema.ResourceData,
 		log.Printf("Deleting schedule group %s", d.Id())
 		proxyResponse, err := proxy.deleteArchitectSchedulegroups(ctx, d.Id())
 		if err != nil {
-			return proxyResponse, diag.Errorf("Failed to delete schedule group %s: %s", d.Id(), err)
+			return proxyResponse, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to delete schedule group %s", d.Id()), proxyResponse)
```
Comment (charliecon) [inline genesyscloud/architect_schedulegroups/resource_genesyscloud_architect_schedulegroups.go:149]: missing err

File: genesyscloud/auth_role/resource_genesyscloud_auth_role.go [hunk 7]
```diff
@@ -28,7 +28,7 @@ func getAllAuthAuthRoles(ctx context.Context, clientConfig *platformclientv2.Con
 
 	roles, proxyResponse, getErr := proxy.getAllAuthRole(ctx)
 	if getErr != nil {
-		return nil, diag.Errorf("Failed to get auth roles: %v %v ", getErr, proxyResponse)
+		return nil, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to get page of roles"), proxyResponse)
```
Comment (charliecon) [inline genesyscloud/auth_role/resource_genesyscloud_auth_role.go:31]: Missing err

File: genesyscloud/architect_flow/data_source_genesyscloud_flow.go (hunk 2 above)
Comment (charliecon) [reply genesyscloud/architect_flow/data_source_genesyscloud_flow.go:27]: There's no need to change all the error messages in this PR to lower case, just saying this as a heads up that changing from lower to upper is the wrong way to go

**Review by charliecon — APPROVED** [review]
Looks good. Still just missing the error object from the string in a few places

**Review by carnellj-genesys — APPROVED** [review]

**Pull request merged**